clients/include/spmm/testing_compress.hpp

```diff
@@ -570,47 +635,78 @@ void testing_compress(const Arguments& arg)
         {
+            order = orderA;
+
+            row      = M;
+            col      = K;
+            stride_1 = stride_1_a;
+            stride_2 = stride_2_a;
+            stride   = stride_a;
+
             //compressd matrix
             c_row      = transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? M : K / 2;
             c_col      = transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? K / 2 : M;
-            c_ld       = transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? M : K / 2;
+            c_ld       = orderA == HIPSPARSE_ORDER_COL ? c_row : c_col;
             c_stride_1 = transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? 1 : c_ld;
             c_stride_2 = transA == HIPSPARSE_OPERATION_NON_TRANSPOSE ? c_ld : 1;
             c_stride_r = K / 2 * M;
             c_stride   = stride_a == 0 ? 0 : c_stride_r;
 
             //metadata
-            m_ld       = M;
-            m_stride_1 = K / 8;
+            m_row      = M;
+            m_col      = K / 8;
+            m_ld       = orderA == HIPSPARSE_ORDER_COL ? m_row : m_col; //M;
+            m_stride_1 = m_col; //K / 8;
             m_stride_2 = 1;
-            m_stride_r = M * m_stride_1;
+            m_stride_r = m_row * m_col;
             m_stride   = stride_a == 0 ? 0 : m_stride_r;
         }
         else
         {
+            order = orderB;
+
+            row      = K;
+            col      = N;
+            stride_1 = stride_1_b;
+            stride_2 = stride_2_b;
+            stride   = stride_b;
+
             //compressd matrix
             c_row      = transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? K / 2 : N;
             c_col      = transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? N : K / 2;
-            c_ld       = transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? K / 2 : N;
+            c_ld       = orderB == HIPSPARSE_ORDER_COL ? c_row : c_col;
             c_stride_1 = transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? 1 : c_ld;
             c_stride_2 = transB == HIPSPARSE_OPERATION_NON_TRANSPOSE ? c_ld : 1;
             c_stride_r = K / 2 * N;
             c_stride   = stride_b == 0 ? 0 : c_stride_r;
 
             //metadata
-            m_ld       = K / 8;
+            m_row      = K / 8;
+            m_col      = N;
+            m_ld       = orderB == HIPSPARSE_ORDER_COL ? m_row : m_col; //K / 8;
             m_stride_1 = 1;
-            m_stride_2 = K / 8;
-            m_stride_r = N * m_stride_2;
+            m_stride_2 = m_row;
+            m_stride_r = m_row * m_col;
             m_stride   = stride_b == 0 ? 0 : m_stride_r;
         }
 
-        auto metadata_offset = c_stride_r * sizeof(Ti)
-                               * ((arg.sparse_b ? stride_b : stride_a) == 0 ? 1 : num_batches);
+        if(order == HIPSPARSE_ORDER_ROW)
+        {
+            std::swap(stride_1, stride_2);
+            std::swap(c_row, c_col);
+            std::swap(c_stride_1, c_stride_2);
+            std::swap(m_row, m_col);
+            //std::swap(m_stride_1, m_stride_2);
```

Comment:
Is this because the **m_stride_2 = m_row**?
m_row and m_col is swapped however m_stride_2 is unchanged.